core/Cargo.toml

```diff
@@ -19,48 +19,27 @@ doctest = false
 
 [features]
 default = ["std"]
-diagnostics = ["dep:miette"]
-serde = ["dep:serde"]
 std = [
-    "dep:parking_lot",
-    "memchr/std",
     "miden-crypto/std",
     "miden-formatting/std",
     "math/std",
-    "serde/std",
     "winter-utils/std",
     "thiserror/std",
 ]
+serde = ["miden-core-diagnostics/serde", "miden-crypto/serde", "math/serde"]
+# DEPRECATED
+diagnostics = []
```

Comment:
Let's remove this - every release comes with breaking changes anyways. Plus, I see that `miden-processor` still needlessly requires this feature (just noise at this point).

Reply:
I think removing this is fine, as @plafer mentioned, we are still introducing very significant breaking changes and I would rather introduce these now than later.